.github/workflows/build_and_test.yml

```diff
@@ -96,12 +99,23 @@ jobs:
         KONG_TEST_PG_DATABASE: kong
         KONG_TEST_PG_USER: kong
       run: |
-          source ${{ env.BUILD_ROOT }}/kong-dev-venv.sh
-          TEST_CMD="bin/busted -v -o htest spec/01-unit"
-          if [[ $KONG_TEST_COVERAGE = true ]]; then
-            TEST_CMD="$TEST_CMD --coverage"
-          fi
-          $TEST_CMD
+        echo "export KONG_TEST_PG_DATABASE=\"$KONG_TEST_PG_DATABASE\"" > run.sh
+        echo "export KONG_TEST_PG_USER=\"$KONG_TEST_PG_USER\"" >> run.sh
+        echo "KONG_TEST_COVERAGE=\"$KONG_TEST_COVERAGE\"" >> run.sh
+        echo "ulimit -c unlimited" >> run.sh
+        echo "mkdir -p /tmp/cores" >> run.sh
+        echo "chmod 777 /tmp/cores" >> run.sh
+        echo "echo '/tmp/cores/core.%e.%p' | tee /proc/sys/kernel/core_pattern" >> run.sh
+        echo "sysctl -w fs.suid_dumpable=2" >> run.sh
+        echo "sysctl -p" >> run.sh
+        echo "source ${{ env.BUILD_ROOT }}/kong-dev-venv.sh" >> run.sh
+        echo "TEST_CMD=\"bin/busted -v -o htest spec/01-unit\"" >> run.sh
+        echo "if [[ \$KONG_TEST_COVERAGE = true ]]; then" >> run.sh
+        echo "  TEST_CMD=\"\$TEST_CMD --coverage\"" >> run.sh
+        echo "fi" >> run.sh
+        echo "\$TEST_CMD" >> run.sh
+
+        sudo bash run.sh
```

Comment:
we can use sudo -E bin/busted -v -o htest spec/01-unit to passthrough env vars without need to >> run.sh